oqsprov/oqsprov.c

```diff
@@ -292,15 +292,17 @@ static const OSSL_PARAM *oqsprovider_gettable_params(void *provctx)
 static int oqsprovider_get_params(void *provctx, OSSL_PARAM params[])
 {
     OSSL_PARAM *p;
+    char buildinfo[512];
 
+    snprintf(buildinfo, 512, "OQS Provider v.%s based on liboqs v.%s", OQS_PROVIDER_VERSION_STR, OQS_VERSION_TEXT);
     p = OSSL_PARAM_locate(params, OSSL_PROV_PARAM_NAME);
     if (p != NULL && !OSSL_PARAM_set_utf8_ptr(p, "OpenSSL OQS Provider"))
         return 0;
     p = OSSL_PARAM_locate(params, OSSL_PROV_PARAM_VERSION);
-    if (p != NULL && !OSSL_PARAM_set_utf8_ptr(p, OPENSSL_VERSION_STR))
+    if (p != NULL && !OSSL_PARAM_set_utf8_ptr(p, OQS_PROVIDER_VERSION_STR))
         return 0;
     p = OSSL_PARAM_locate(params, OSSL_PROV_PARAM_BUILDINFO);
-    if (p != NULL && !OSSL_PARAM_set_utf8_ptr(p, OPENSSL_FULL_VERSION_STR))
+    if (p != NULL && !OSSL_PARAM_set_utf8_ptr(p, buildinfo))
```

Comment:
I think OSSL_PARAM_set_utf8_ptr just sets a pointer, and `buildinfo` would not be valid anymore after the function returns.

Reply:
Hmm, strdup malloc's memory and I suspect this way it will lead to a memory leak.

Couldn't you assemble the build info string with a #define? No runtime info seems to be needed to do this. Then it can be safely set with OSSL_PARAM_set_utf8_ptr.
I interpret the documentation that it really just sets the pointer: https://www.openssl.org/docs/manmaster/man3/OSSL_PARAM_utf8_ptr.html

Reply:
True. Done in https://github.com/open-quantum-safe/oqs-provider/pull/45/commits/7a3536ef70cc295e02cb2614cc3b15a70d5d720c .